<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>linkSet • linkSet</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="linkSet">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">linkSet</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/linkSet.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/GilbertHan1011/linkSet/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>linkSet</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/GilbertHan1011/linkSet/blob/HEAD/vignettes/linkSet.Rmd" class="external-link"><code>vignettes/linkSet.Rmd</code></a></small>
      <div class="d-none name"><code>linkSet.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>With techniche like HiC, PCHiC, scATAC-seq, and other high-throughput
omics techniques, we are able to predict the interactions between genes
and enhancers. The linkSet package provides classes to represent the
interactions between genes and enhancers, and provides functions to
analyze and visualize the interactions.</p>
</div>
<div class="section level2">
<h2 id="the-linkset-class">The linkSet class<a class="anchor" aria-label="anchor" href="#the-linkset-class"></a>
</h2>
<p>The linkSet class is a specialized data structure designed to
represent and analyze genomic interactions, particularly focusing on
gene-enhancer relationships. It’s part of the linkSet package, which
provides tools for managing and analyzing genomic link data in
bioinformatics and genomic research.</p>
<p>Key features of the linkSet class:</p>
<ol style="list-style-type: decimal">
<li>Representation of genomic interactions:</li>
</ol>
<ul>
<li>It stores two types of genomic anchors: “bait” (typically genes) and
“other end” (oe, typically enhancers or other regulatory elements).</li>
<li>These anchors are represented as genomic ranges, allowing for
precise localization on chromosomes.</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Flexible input: Can be constructed from various data types,
including GRanges objects for both anchors, or character vectors for
bait and GRanges for other ends. Supports conversion from other common
genomic data structures like GInteractions and data frames.</li>
<li>Metadata storage: Allows for additional information to be stored
alongside the genomic interactions, such as interaction scores or
experimental conditions.</li>
<li>Biological context: Designed to work with data from various
high-throughput genomic techniques like Hi-C, Promoter Capture Hi-C
(PCHi-C), and scATAC-seq. Facilitates the analysis of long-range
chromatin interactions, which are crucial for understanding gene
regulation and 3D genome organization.</li>
<li>Annotation capabilities: Includes methods for annotating promoters
and distinguishing between inter- and intra-chromosomal interactions.
This feature is particularly useful for identifying potential regulatory
relationships between distal genomic elements.</li>
<li>Integration with Bioconductor: Built on top of established
Bioconductor classes, ensuring compatibility with a wide range of
genomic analysis tools.</li>
<li>Visualization and analysis: The class is designed to support
functions for analyzing and visualizing genomic interactions.</li>
</ol>
<p><img src="img/overview.png" alt="Overview diagram of the linkSet class structure and its components" width="70%"></p>
</div>
<div class="section level2">
<h2 id="construction">Construction<a class="anchor" aria-label="anchor" href="#construction"></a>
</h2>
<p>The linkSet class can be constructed from various data types,
including GRanges objects for both anchors, or character vectors for
bait and GRanges for other ends.</p>
<div class="section level3">
<h3 id="construction-from-granges-objects">Construction from GRanges objects<a class="anchor" aria-label="anchor" href="#construction-from-granges-objects"></a>
</h3>
<p>To construct a <code>linkSet</code> object from GRanges objects:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/GilbertHan1011/linkSet" class="external-link">linkSet</a></span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span></span>
<span><span class="va">gr1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr1"</span>, <span class="st">"chr2"</span><span class="op">)</span>,</span>
<span>              ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">100</span>, <span class="fl">200</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>              strand <span class="op">=</span> <span class="st">"+"</span>, symbol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Gene1"</span>, <span class="st">"Gene2"</span>, <span class="st">"Gene3"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">gr2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"chr2"</span><span class="op">)</span>,</span>
<span>              ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">50</span>, <span class="fl">150</span>, <span class="fl">250</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>              strand <span class="op">=</span> <span class="st">"+"</span><span class="op">)</span></span>
<span><span class="co">## construct linkSet</span></span>
<span><span class="va">ls</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/linkSet.html">linkSet</a></span><span class="op">(</span><span class="va">gr1</span>, <span class="va">gr2</span>, specificCol <span class="op">=</span> <span class="st">"symbol"</span><span class="op">)</span></span>
<span><span class="va">ls</span></span>
<span><span class="co">#&gt; linkSet object with 3 interactions and 1 metadata column:</span></span>
<span><span class="co">#&gt;              bait     seqnames_oe ranges_oe | anchor1.symbol</span></span>
<span><span class="co">#&gt;       &lt;character&gt;           &lt;Rle&gt; &lt;IRanges&gt; |    &lt;character&gt;</span></span>
<span><span class="co">#&gt;   [1]       Gene1 ---        chr1     50-59 |          Gene1</span></span>
<span><span class="co">#&gt;   [2]       Gene2 ---        chr2   150-159 |          Gene2</span></span>
<span><span class="co">#&gt;   [3]       Gene3 ---        chr2   250-259 |          Gene3</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   regions: 3 ranges and 0 metadata columns</span></span>
<span><span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="construction-from-ginteractions">Construction from GInteractions<a class="anchor" aria-label="anchor" href="#construction-from-ginteractions"></a>
</h3>
<p>You can also create a <code>linkSet</code> object from a
<code>GInteractions</code> object using the <code>Convert</code>
function:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">InteractionSet</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'InteractionSet'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:linkSet':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anchorIds, anchors, pairdist, regions</span></span>
<span><span class="va">gi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/InteractionSet/man/GInteractions-class.html" class="external-link">GInteractions</a></span><span class="op">(</span></span>
<span>  anchor1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">5</span><span class="op">)</span>,</span>
<span>  anchor2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">4</span>, <span class="fl">6</span><span class="op">)</span>,</span>
<span>  regions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span></span>
<span>    seqnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"chr2"</span>, <span class="st">"chr3"</span>, <span class="st">"chr3"</span><span class="op">)</span>,</span>
<span>    ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">50</span>, <span class="fl">100</span>, <span class="fl">150</span>, <span class="fl">200</span>, <span class="fl">250</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">gi</span><span class="op">)</span><span class="op">$</span><span class="va">score</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.2</span>, <span class="fl">0.3</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">gi</span><span class="op">)</span><span class="op">$</span><span class="va">gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Sox2"</span>, <span class="st">"Sox9"</span>, <span class="st">"Sox10"</span><span class="op">)</span></span>
<span><span class="co">##Convert to linkSet</span></span>
<span><span class="va">ls_from_gi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Convert.html">Convert</a></span><span class="op">(</span><span class="va">gi</span>, baitCol <span class="op">=</span> <span class="st">"gene"</span><span class="op">)</span></span>
<span><span class="va">ls_from_gi</span></span>
<span><span class="co">#&gt; linkSet object with 3 interactions and 2 metadata columns:</span></span>
<span><span class="co">#&gt;              bait     seqnames_oe ranges_oe |     score        gene</span></span>
<span><span class="co">#&gt;       &lt;character&gt;           &lt;Rle&gt; &lt;IRanges&gt; | &lt;numeric&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;   [1]        Sox2 ---        chr1     50-59 |       0.1        Sox2</span></span>
<span><span class="co">#&gt;   [2]        Sox9 ---        chr2   150-159 |       0.2        Sox9</span></span>
<span><span class="co">#&gt;   [3]       Sox10 ---        chr3   250-259 |       0.3       Sox10</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   regions: 3 ranges and 0 metadata columns</span></span>
<span><span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>Some times, you need to use gene and enhancer information to
construct a <code>linkSet</code> object from GInteractions.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geneGr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span></span>
<span>  seqnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"chr3"</span><span class="op">)</span>,</span>
<span>  ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">105</span>, <span class="fl">205</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>,</span>
<span>  symbol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Gene1"</span>, <span class="st">"Gene2"</span>, <span class="st">"Gene3"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">peakGr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span></span>
<span>  seqnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"chr3"</span><span class="op">)</span>,</span>
<span>  ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">45</span>, <span class="fl">145</span>, <span class="fl">245</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run baitGInteractions</span></span>
<span><span class="va">ls_from_gi_2</span> <span class="op">&lt;-</span> <span class="fu">baitGInteractions</span><span class="op">(</span><span class="va">gi</span>, <span class="va">geneGr</span>, <span class="va">peakGr</span>, geneSymbol <span class="op">=</span> <span class="st">"symbol"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ls_from_gi_2</span></span>
<span><span class="co">#&gt; linkSet object with 3 interactions and 3 metadata columns:</span></span>
<span><span class="co">#&gt;              bait     seqnames_oe ranges_oe | anchor1.symbol     score</span></span>
<span><span class="co">#&gt;       &lt;character&gt;           &lt;Rle&gt; &lt;IRanges&gt; |    &lt;character&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   [1]       Gene1 ---        chr1     45-54 |          Gene1       0.1</span></span>
<span><span class="co">#&gt;   [2]       Gene2 ---        chr2   145-154 |          Gene2       0.2</span></span>
<span><span class="co">#&gt;   [3]       Gene3 ---        chr3   245-254 |          Gene3       0.3</span></span>
<span><span class="co">#&gt;              gene</span></span>
<span><span class="co">#&gt;       &lt;character&gt;</span></span>
<span><span class="co">#&gt;   [1]        Sox2</span></span>
<span><span class="co">#&gt;   [2]        Sox9</span></span>
<span><span class="co">#&gt;   [3]       Sox10</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   regions: 3 ranges and 0 metadata columns</span></span>
<span><span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="other-construction-methods">Other construction methods<a class="anchor" aria-label="anchor" href="#other-construction-methods"></a>
</h3>
<p>You can also construct a <code>linkSet</code> object from a data
frame.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene1"</span>, <span class="st">"gene2"</span>, <span class="st">"gene3"</span><span class="op">)</span>,</span>
<span>  peak <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1:1000-2000"</span>, <span class="st">"chr2:3000-4000"</span>, <span class="st">"chr3:5000-6000"</span><span class="op">)</span>,</span>
<span>  score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">0.7</span>, <span class="fl">0.9</span><span class="op">)</span>,</span>
<span>  stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Test successful conversion</span></span>
<span><span class="va">ls_from_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Convert.html">Convert</a></span><span class="op">(</span><span class="va">test_df</span><span class="op">)</span></span>
<span><span class="va">ls_from_df</span></span>
<span><span class="co">#&gt; linkSet object with 3 interactions and 1 metadata column:</span></span>
<span><span class="co">#&gt;              bait     seqnames_oe ranges_oe |     score</span></span>
<span><span class="co">#&gt;       &lt;character&gt;           &lt;Rle&gt; &lt;IRanges&gt; | &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   [1]       gene1 ---        chr1  999-2000 |       0.5</span></span>
<span><span class="co">#&gt;   [2]       gene2 ---        chr2 2999-4000 |       0.7</span></span>
<span><span class="co">#&gt;   [3]       gene3 ---        chr3 4999-6000 |       0.9</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   regions: 3 ranges and 0 metadata columns</span></span>
<span><span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="accessors">Accessors<a class="anchor" aria-label="anchor" href="#accessors"></a>
</h2>
<div class="section level3">
<h3 id="getters">Getters<a class="anchor" aria-label="anchor" href="#getters"></a>
</h3>
</div>
</div>
<div class="section level2">
<h2 id="subsetting">Subsetting<a class="anchor" aria-label="anchor" href="#subsetting"></a>
</h2>
</div>
<div class="section level2">
<h2 id="distance-calculation">Distance calculation<a class="anchor" aria-label="anchor" href="#distance-calculation"></a>
</h2>
</div>
<div class="section level2">
<h2 id="annotations">Annotations<a class="anchor" aria-label="anchor" href="#annotations"></a>
</h2>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Gilbert Han.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
